other/text_to_speech.py

```diff
@@ -0,0 +1,119 @@
+"""
+Created by: Ariel Sperduti
+
+Given a text it converts into an  MP3 audio file.
+Under the hood it uses fromtexttospeech.com.
+
+Example:
+    text_to_speech(text="¡Hola Mundo!", language="Spanish", voice="Mateo")
+"""
+
+import re
+import subprocess
+import urllib.parse
+
+import requests
+
+LANGUAGES_VOICES = {
+    "US English": ["Alice", "Daisy", "George", "Jenna", "John"],
+    "British": ["Emma", "Harry"],
+    "French": ["Gabriel", "Jade"],
+    "Spanish": ["Isabella", "Mateo"],
+    "German": ["Michael", "Nadine"],
+    "Italian": ["Alessandra", "Giovanni"],
+    "Portuguese": ["Rodrigo"],
+    "Russian": ["Valentina"],
+}
+
+VOICES = {
+    "Emma": "IVONA Amy22 (UK English)",
+    "Harry": "IVONA Brian22 (UK English)",
+    "Jade": "IVONA CΘline22 (French)",
+    "Gabriel": "IVONA Mathieu22 (French)",
+    "Nadine": "IVONA Marlene22 (German)",
+    "Michael": "IVONA Hans22 (German)",
+    "Valentina": "IVONA Tatyana22 (Russian)",
+    "John": "IVONA Eric22",
+    "Jenna": "IVONA Jennifer22",
+    "George": "IVONA Joey22",
+    "Alice": "IVONA Kimberly22",
+    "Daisy": "IVONA Salli22",
+    "Alessandra": "IVONA Carla22 (Italian)",
+    "Giovanni": "IVONA Giorgio22 (Italian)",
+    "Isabella": "IVONA Conchita22 (Spanish [Modern])",
+    "Mateo": "IVONA Enrique22 (Spanish [Modern])",
+    "Rodrigo": "IVONA Cristiano22 (Portuguese)",
+}
+
+
+def text_to_speech(
+    text: str = "Hello world",
+    language: str = "British",
+    voice: str = "Emma",
+    audio_path: str = "output.mp3",
+) -> bool:
+
+    if language not in LANGUAGES_VOICES:
+        raise ValueError(f"Language '{language}' not available")
+
+    if voice not in LANGUAGES_VOICES[language]:
+        raise ValueError(f"Voice '{voice}' not available")
+
+    # prepare data
+    text_encoded = urllib.parse.quote(text.encode("utf8"))
+    data = f"input_text={text_encoded}&language={language}\
+            &voice={VOICES[voice]}&speed=0&action=process_text"
+
+    # make request to www.fromtexttospeech.com
+    # it gives an html with the mp3's url
+    args = [
```

Comment:
This is a Python algorithm repository so it makes more sense to use the `requests` library instead of `curl`. Please make the necessary changes.

Here's the reference to `requests` docs: https://requests.readthedocs.io/en/master/

Reply:
You are right, also curl is not available by default on Windows (Among all the PCs in world, Windows account for 88%, this code will be compatible for 12% people only)

Reply:
> This is a Python algorithm repository so it makes more sense to use the `requests` library instead of `curl`. Please make the necessary changes.
> 
> Here's the reference to `requests` docs: https://requests.readthedocs.io/en/master/
HI @dhruvmanila , yes you are right. I know about requests, I use it in the script to download the mp3. But I was not able to implement the download of the page with requests, even using the same headers as in curl. I will give a new view.

Reply:
Please take a look at all the parameters accepted by the `request` method: https://requests.readthedocs.io/en/latest/api/#requests.request

All your headers (`-H`) should be provided in a dictionary to the `headers` parameter. All your data (`-d`) should also be in a dictionary to the `data` parameter. 

As you can see, it is pretty intuitive to understand. In a similar way, you can construct your other request parameters.